UI/src/app/services/common-utils.service.ts

```diff
@@ -44,4 +49,27 @@ export class CommonUtilsService {
       data: {action, target: this.stateService.Target}
     });
   }
+
+  openResetWarningDialog() {
+    const matDialog = this.matDialog.open(ResetWarningComponent, {
+      closeOnNavigation: false,
+      disableClose: false,
+      panelClass: 'warning-dialog',
+    });
+
+    matDialog.afterClosed().pipe(
```

Comment:
I think you have no need in pipe and map in this case. Just do the same in subscribe

Reply:
Done!